source/collections/Queue.ooc

```diff
@@ -89,4 +89,8 @@ VectorQueue: class <T> extends Queue<T> {
 		this _capacity = newCapacity
 	}
 	toString: func -> String { "Queue: count=" << this _count toString() >> " capacity=" & this _capacity toString() }
+	operator [] (index: Int) -> T {
+		position := index >= 0 ? this _head + index : this _tail + index
+		this _backend[position % this _capacity]
```

Comment:
To reiterate, you need to use `Int modulo`, because position be a negative number.